src/scale.ts

```diff
@@ -51,6 +50,9 @@ const SCALE_CATEGORY_INDEX: {
   log: 'numeric',
   pow: 'numeric',
   sqrt: 'numeric',
+  quantize: 'numeric',
```

Comment:
linear/log/pow (numeric scales) map  Q => Q values

quantile, quantize, threshold map quantitative to discrete values and thus shouldn't be the same scale type. 

If you want to support these, I suggest that you separate adding them to another PR first since there are probably delicate details that requires thorough review.

Reply:
quantile, quantize, threshold (and bin-linear, bin-ordinal) are called discretizing scales in Vega https://vega.github.io/vega/docs/scales/#discretizing

Reply:
I think we maybe want to leave if bin-linear and bin-ordinal using their keyword though as we probably don't want to allow merging them?